xwiki-rendering-legacy/xwiki-rendering-legacy-transformations/xwiki-rendering-legacy-transformation-macro/src/main/aspect/org/xwiki/rendering/macro/descriptor/AbstractMacroDescriptorAspect.aj

```diff
@@ -0,0 +1,24 @@
+package org.xwiki.rendering.macro.descriptor;
+
+import java.util.Set;
+
+/**
+ * Add a backward compatibility layer to the {@link AbstractMacroDescriptor} class.
+ *
+ * @version $Id$
+ * @since 14.6RC1
+ */
+public privileged aspect AbstractMacroDescriptorAspect
+{
+    /**
+     * @param defaultCategory default category under which this macro should be listed
+     * @see MacroDescriptor#getDefaultCategories()
+     * @deprecated since 14.6RC1 use {@link AbstractMacroDescriptor#setDefaultCategories(String...)} or {@link  AbstractMacroDescriptor#setDefaultCategories(Set)}
```

Comment:
Stupid question: what is a non-default category?

Reply:
This is a list of categories before they are overridden. The non-default categories can be defined using `xwiki.properties` or the Admin UI. 
Note that while `org.xwiki.rendering.internal.macro.DefaultMacroCategoryManager#getMacroIdsByCategory` takes the override into account, to my knowledge no code visible to the end users currently take the overriding into account.